### DISCUSSION TITLE
Embed videos in markdown?

### DISCUSSION THREAD

**Discussion by tettoffensive** [remarkjs/remark#1043] [Q&A] [answered]
I'm looking to embed vimeo videos in my markdown, which currently involves using rehype-raw which is a big library and it's also the only thing I need it for. I came across [this gatsby plugin](https://github.com/ahmadawais/gatsby-remark-better-embed-video), but I'm not using gatsby. Just using ReactMarkdown with NextJS.

Are there any alternative solutions out there? Or would it be simple to convert the gatsby plugin to a regular remark plugin?

**Comment by ChristianMurphy** [comment, marked as answer]
There are a few options out there in plain markdown.
See the previous discussions on video in markdown for those options:
* https://github.com/orgs/remarkjs/discussions/957
* https://github.com/orgs/remarkjs/discussions/557
* https://github.com/orgs/remarkjs/discussions/747

It is also worth nothing Next supports MDX https://nextjs.org/docs/advanced-features/using-mdx (which builds JSX into markdown/remark)
Which would allow the `<video>` tag or any react video player to be used.

**Reply by ChristianMurphy** [reply]
A new plugin matching the Gatsby implementation could also be made, this page gives an overview of working with ASTs and creating plugins https://unifiedjs.com/learn/
